README.md
# HIJRI CALENDAR

[![Build Status](https://travis-ci.org/hmtanbir/hijricalendar.svg?branch=master)](https://travis-ci.org/hmtanbir/hijricalendar)
[![Version](https://img.shields.io/pypi/v/hijricalendar.svg?)](http://badge.fury.io/py/hijricalendar)
[![Python](https://img.shields.io/pypi/pyversions/hijricalendar.svg?)](https://pypi.python.org/pypi/hijricalendar/0.0.1)
[![Size](https://img.shields.io/github/size/hmtanbir/hijricalendar/hijricalendar/__init__.py.svg?)](https://github.com/hmtanbir/hijricalendar/)
[![codecov](https://codecov.io/gh/hmtanbir/hijricalendar/branch/master/graph/badge.svg)](https://codecov.io/gh/hmtanbir/hijricalendar)

Hijri Calendar is a package for Bangla language users with various functionalities including Arabic date in bangla and english format.

It can be used to get Arabic date that includes year, month, date, weekday in Bangla language and 
English language.

Furthermore, It can be used in global wide. Day difference between
arabic country and south asia country is considered in this module. 

This software can be used on Linux/Unix, Mac OS and Windows systems.

# Features


Get Bangla date that includes:

   - Bangla Hijri Date (১-৩০)

   - Bangla Hijri Month ('মহরম','সফর','রবিউল আউয়াল','রবিউস সানি','জুমাদিউল আউয়াল','জুমাদিউস সানি','রজব','শাবান','রমজান','শাওয়াল','জিলক্বাদ','জিলহাজ্জ্ব')

   - Bangla Hijri Year (১৪৩৯ - )
   
   - English Hijri Date (1-30)

   - English Hijri Month ('Muharram', 'Safar', 'Rabi-ul-Awwal', 'Rabi-us-Sani', 'Jamadi-ul-Awwal', 'Jamadi-us-Sani', 'Rajab', 'Shaban', 'Ramadan', 'Shawwal','Jil-Quadah', 'Jil-Hajjah',)

   - English Hijri Year (1439 - )    



# Installation


We recommend install ``hijricalendar``  through pip install using Python 3 but it supports
python 2.7

Run this command when you are using virtualenv.
 ```
 $ pip install hijricalendar
 ```
 For global(when you are not using virtualenv)

 ```
 $ sudo pip install hijricalendar
 ```
 
# Example
Open python shell:

```
$ python
```
Output:
```
Python 2.7.12 (default, Dec  4 2017, 14:50:18) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>>

```


To get today's date in Bangla Hijri calendar:

```
$ import hijricalendar
 
$ bangla_date = get_date()

$ print bangla_date
```

``Output: {'date': '৪', 'year': '১৪৩৯', 'weekday': 'বুধবার', 'month': '৬'}``

```
$ date = bangla_date['date']

$ year = bangla_date['year']

$ weekday = bangla_date['weekday']

$ month = bangla_date['month']

$ print date, year, weekday, month
```
 
``Output:   ৪ ১৪৩৯ বুধবার ৬``

To get today's date in English Hijri calendar:

``` 
 $ en_date = get_date(en=True)
 $ print en_date
```

``Output:   {'date': 4, 'year': 1439, 'weekday': 'Wednesday', 'month': 6}``

```
 $ date = en_date['date']
 
 $ year = en_date['year']
 
 $ weekday = en_date['weekday']
 
 $ month = en_date['month']
 
 $ print date, year, weekday, month
```

``Output:   4 1439 Wednesday 6``

To get today's date in Bangla Hijri calendar in arabic timezone:

```
$ import hijricalendar
 
$ bangla_date = get_date(arabic=True)

$ print bangla_date
```

``Output: {'date': '৫', 'year': '১৪৩৯', 'weekday': 'বুধবার', 'month': '৬'}``

```
$ date = bangla_date['date']

$ year = bangla_date['year']

$ weekday = bangla_date['weekday']

$ month = bangla_date['month']

$ print date, year, weekday, month
```
 
``Output:  ৫  ১৪৩৯ বুধবার ৬``

To get today's date in English Hijri calendar in arabic timezone:

``` 
 $ en_date = get_date(en=True)
 $ print en_date
```

``Output:   {'date': 5, 'year': 1439, 'weekday': 'Wednesday', 'month': 6}``

```
 $ date = en_date['date']
 
 $ year = en_date['year']
 
 $ weekday = en_date['weekday']
 
 $ month = en_date['month']
 
 $ print date, year, weekday, month
```

``Output:   5 1439 Wednesday 6``

To get specific date in Bangla Hijri calendar:

```
# format: get_date(date, month, year)
 
$ bangla_date = get_date(21, 2, 2018)

$ print bangla_date
```

``Output: {'date': '৪', 'year': '১৪৩৯', 'weekday': 'বুধবার', 'month': '৬'}``

```
$ date = bangla_date['date']

$ year = bangla_date['year']

$ weekday = bangla_date['weekday']

$ month = bangla_date['month']

$ print date, year, weekday, month
```
 
``Output:   ৪ ১৪৩৯ বুধবার ৬``

To get specific date in English Hijri calendar:

```
 # format: get_date(date, month, year, en=True)
 $ en_date = get_date(21, 2, 2018, en=True)
 $ print en_date
```

``Output:   {'date': 4, 'year': 1439, 'weekday': 'Wednesday', 'month': 6}``

```
 $ date = en_date['date']
 
 $ year = en_date['year']
 
 $ weekday = en_date['weekday']
 
 $ month = en_date['month']
 
 $ print date, year, weekday, month
```

``Output:   4 1439 Wednesday 6``

# Contribute


Create Github Pull Request https://github.com/hmtanbir/hijricalendar/pulls


If you have suggestion use GitHub issue system or send a message in linkedin https://www.linkedin.com/in/hmtanbir.


changes.txt
Changes
=======

0.0.1 (2018-02-21)
------------------
- Initial release

tests/unit_tests.py
# -*- coding: utf-8 -*-
import unittest
import unittest
import os
import sys
import bangla
from umalqurra.hijri_date import HijriDate

BASEDIR = os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), ".."))
sys.path.insert(0, BASEDIR)

import hijricalendar


class TestHijriCalendar(unittest.TestCase):
    def test_get_date(self):
        today = HijriDate.today()
        date = bangla.convert_english_digit_to_bangla_digit(int(today.day) - int(1))
        year = bangla.convert_english_digit_to_bangla_digit(int(today.year))
        month = bangla.convert_english_digit_to_bangla_digit(int(today.month))
        weekday = hijricalendar.weekdays_dict[today.day_name_en]

        result = hijricalendar.get_date()
        expect = {'date': date, 'year': year, 'weekday': weekday, 'month': month}
        self.assertEquals(result, expect)

    def test_get_date_when_en_is_true(self):
        today = HijriDate.today()
        date = int(today.day) - int(1)
        year = int(today.year)
        month = int(today.month)
        weekday = today.day_name_en

        result = hijricalendar.get_date(en=True)
        expect = {'date': date, 'year': year, 'weekday': weekday, 'month': month}
        self.assertEquals(result, expect)

    def test_get_date_with_args(self):
        result = hijricalendar.get_date(21, 2, 2018)
        expect = {'date': '৪', 'year': '১৪৩৯', 'weekday': 'বুধবার', 'month': '৬'}
        self.assertEquals(result, expect)

    def test_get_date_with_args_when_only_arabic_is_true(self):
        result = hijricalendar.get_date(21, 2, 2018, arabic=True)
        expect = {'date': '৫', 'year': '১৪৩৯', 'weekday': 'বুধবার', 'month': '৬'}
        self.assertEquals(result, expect)

    def test_get_date_with_args_when_en_is_true(self):
        result = hijricalendar.get_date(21, 2, 2018, en=True)
        expect = {'date': 4, 'year': 1439, 'weekday': 'Wednesday', 'month': 6}
        self.assertEquals(result, expect)

    def test_get_date_with_args_when_both_en_and_arabic_is_true(self):
        result = hijricalendar.get_date(21, 2, 2018, en=True, arabic=True)
        expect = {'date': 5, 'year': 1439, 'weekday': 'Wednesday', 'month': 6}
        self.assertEquals(result, expect)

    def test_get_month(self):
        result = hijricalendar.get_month()
        today = HijriDate.today()
        month = int(today.month)
        expect = month
        self.assertEquals(result, expect)

    def test_get_month_for_specific_date(self):
        result = hijricalendar.get_month(21, 2, 2018)
        expect = 6
        self.assertEquals(result, expect)

    def test_get_month_name(self):
        result = hijricalendar.get_month_name(21, 2, 2018)
        expect = hijricalendar.month_name_dict_bn[6]  # since arabic month =6 using get_month method
        self.assertEquals(result, expect)

    def test_get_month_name_when_en_is_true(self):
        result = hijricalendar.get_month_name(21, 2, 2018, en=True)
        expect = hijricalendar.month_name_dict_en[6]  # since arabic month =6 using get_month method
        self.assertEquals(result, expect)


if __name__ == "__main__":
    unittest.main()


hijricalendar/__init__.py
# -*- coding: utf-8 -*-

import bangla
from umalqurra.hijri_date import HijriDate

bangla_weekdays = bangla.python_bangla_weekdays

month_name_dict_en = {
    1: 'Muharram',
    2: 'Safar',
    3: 'Rabi-ul-Awwal',
    4: 'Rabi-us-Sani',
    5: 'Jamadi-ul-Awwal',
    6: 'Jamadi-us-Sani',
    7: 'Rajab',
    8: 'Shaban',
    9: 'Ramadan',
    10: 'Shawwal',
    11: 'Jil-Quadah',
    12: 'Jil-Hajjah',
}

month_name_dict_bn = {
    1: 'মহরম',
    2: 'সফর',
    3: 'রবিউল আউয়াল',
    4: 'রবিউস সানি',
    5: 'জুমাদিউল আউয়াল',
    6: 'জুমাদিউস সানি',
    7: 'রজব',
    8: 'শাবান',
    9: 'রমজান',
    10: 'শাওয়াল',
    11: 'জিলক্বাদ',
    12: 'জিলহাজ্জ্ব',
}

weekdays_dict = {
    'Monday': bangla_weekdays[0],
    'Tuesday': bangla_weekdays[1],
    'Wednesday': bangla_weekdays[2],
    'Thursday': bangla_weekdays[3],
    'Friday': bangla_weekdays[4],
    'Saturday': bangla_weekdays[5],
    'Sunday': bangla_weekdays[6],
}


# Hijri date in bangla format
# Example: $ bangla_date = get_date()
#          $ print bangla_date
#
# Output: {'date': '৪', 'year': '১৪৩৯', 'weekday': 'বুধবার', 'month': '৬'}
#
#           $ date = bangla_date['date']
#           $ year = bangla_date['year']
#           $ weekday = bangla_date['weekday']
#           $ month = bangla_date['month']
#           $ print date, year, weekday, month
#
# Output:   ৪ ১৪৩৯ বুধবার ৬

# Example: $ en_date = get_date(en=True)
#          $ print en_date
#
# Output:   {'date': 4, 'year': 1439, 'weekday': 'Wednesday', 'month': 6}
#
#           $ date = en_date['date']
#           $ year = en_date['year']
#           $ weekday = en_date['weekday']
#           $ month = en_date['month']
#           $ print date, year, weekday, month
#
# Output:   4 1439 Wednesday 6


def get_date(passed_date=None, passed_month=None, passed_year=None, en=False, arabic=False):
    if passed_date == None and passed_month == None and passed_year == None:
        current_date_object = HijriDate.today()
    else:
        current_date_object = HijriDate(passed_year, passed_month, passed_date, gr=True)

    passed_year = int(current_date_object.year)
    passed_month = int(current_date_object.month)

    if arabic == 1:
        passed_date = int(current_date_object.day)
    else:
        passed_date = int(current_date_object.day) - int(1)

    passed_week = current_date_object.day_name_en

    hijri_date_month_year_bn = {
        'date': bangla.convert_english_digit_to_bangla_digit(passed_date),
        'month': bangla.convert_english_digit_to_bangla_digit(passed_month),
        'year': bangla.convert_english_digit_to_bangla_digit(passed_year),
        'weekday': weekdays_dict[passed_week],
    }

    hijri_date_month_year_en = {
        'date': passed_date,
        'month': passed_month,
        'year': passed_year,
        'weekday': passed_week,
    }

    if en == 1:
        return hijri_date_month_year_en
    else:
        return hijri_date_month_year_bn


# return month number in integer format
# $ print get_month()
# output: 6


def get_month(passed_date=None, passed_month=None, passed_year=None):
    if passed_date == None and passed_month == None and passed_year == None:
        current_date_object = HijriDate.today()
    else:
        current_date_object = HijriDate(passed_year, passed_month, passed_date, gr=True)

    return int(current_date_object.month)


# return month name in bangla format
# $ print get_month_name()
# output: জুমাদিউস সানি

# return month name in english format
# $ print get_month_name(en=True)
# output: Jamadi-us-Sani


def get_month_name(passed_date=None, passed_month=None, passed_year=None, en=False):
    month = get_month(passed_date, passed_month, passed_year)

    if en == 1:
        return month_name_dict_en[month]
    else:
        return month_name_dict_bn[month]
